Python Tutorial Exercises(Week 5)/#3.py

```diff
@@ -0,0 +1,14 @@
+'''
+Extract all numbers from a string text and print the sum of it using list comprehension. 
+Ignore punctuation (commas) by use for loop and if i !="," after find them then use .join()
+///if want to remove commas and full stops you can use: if i !="," and i != ".".///
+joins them into a single string. Now you get a string without commas.
+use .isdigit() to find all the numbers in the string then 
+use sum to add the Number list:[2018, 1000000, 2019, 5, 2020, 2] all together 
+then print the total number out.
+'''
+my_string = "Back in the early 2010s, Andrew has always had a keen interest in cars. So much that in 2018, he spent over 1000000 in cash just to buy his favorite car. In 2019, he sold that car just to buy 5 other cars. However, when covid-19 hit NZ in 2020, he had to sell all but 2 of his cars."
+
+print(sum([int(s) for s in ("".join([i for i in my_string if i != ","and i != "."])).split() if s.isdigit()]))
+#Solution 2
+print(sum([int(s) for s in [i.rstrip(",").rstrip(",") for i in my_string.split()] if s.isdigit()]))
```

Comment:
The rstrip() method in your if condition along with isdigit()? It still will have commas and full stops with the numbers.

Reply:
So the .rstrip(",.") method have be to runned before the .isdigit() method. I think put .rstrip(",.") after I is a better idea(in the last loop), because in the next for loop it can have a clean list to find if there have any numbers in that list.  
So I have tried to use ' if number_list1.rstrip(",.").isdigit(): ' but it didn't work, did I do it wrong?

Reply:
**rstrip()**

Reply:
### rstrip()